.github/actions/setup-rust/action.yml

```diff
@@ -30,4 +30,6 @@ runs:
       if: inputs.enable-cache == 'true'
       uses: Swatinem/rust-cache@v2
       with:
-        prefix-key: ${{ inputs.cache-prefix }}
\ No newline at end of file
+        prefix-key: ${{ inputs.cache-prefix }}-${{ inputs.toolchain }}
```

Comment:
Prevent clashes when we add a new toolchain. See https://github.com/o1-labs/openmina/issues/1195